docs/setuptools.txt

```diff
@@ -2293,6 +2293,48 @@ Metadata and options are set in the config sections of the same name.
 
 * Unknown keys are ignored.
 
+setup.cfg-only projects
+=======================
+
+.. versionadded:: 40.9.0
+
+``setuptools`` emulates a dummy ``setup.py`` module containing only
+a ``setuptools.setup()`` call if ``setup.py`` is missing in a
+project directory and PEP 517 build backend is invoked.
+
+.. note::
+
+    PEP 517 doesn't support editable installs so this is currently
+    incompatible with ``pip install --no-build-isolation -e .``.
+
+This means that you can have a Python distribution project with no
+``setup.py`` but having only ``setup.cfg`` present, given you **can rely
+on** having ``setuptools >= 40.9.0`` being set as the build dependency
+and invoked via ``setuptools.build_meta`` PEP 517 backend interface.
+
+To use this feature:
+
+* Specify build requirements and PEP 517 build backend in
+  ``pyproject.toml``.
+  For example:
+
+  .. code-block:: toml
+
+      [build-system]
+      requires = [
+        "setuptools >= 40.9.0",
+        "wheel",
+      ]
+      build-backend = "setuptools.build_meta"
+
+* Use one of PEP 517 compatible build front-ends. Like ``pip >= 19``,
+  ``pep517`` or similar.
+
+  .. warning::
+
+      Beware that due to the fact that PEP517 is new, frontends may have bugs you
```

Comment:
FTR @pganssle and I have agreed that he'll polish the grammar and merge this later.